### PR TITLE
Add SoftwareComponent, Artifact and Publication

### PR DESCRIPTION
**Pull request by martinbonnin**

### CODE REVIEW

File: README.asciidoc [hunk 1]
```diff
@@ -115,6 +115,13 @@ A `settings.gradle` script. A https://docs.gradle.org/current/javadoc/org/gradle
 has a large number of responsibilities, but one of the most important is declaring the set of <<project,projects>>
 that are part of the build, via `include ':project1'` and so on.
 
+[[software-component]]
+=== SoftwareComponent
+A https://docs.gradle.org/current/javadoc/org/gradle/api/component/SoftwareComponent.html[SoftwareComponent] is a list of artifacts
+built by Gradle. It's a relatively recent API used to connect outgoing configurations and publications. Most of the times, you will
```
Comment (autonomousapps) [inline README.asciidoc]: I think we should link `outgoing configurations` to your newly updated definition, and I think we could also use an entry for "publication", as I observed in another review.

Comment (martinbonnin) [reply README.asciidoc]: PR updated to include `Publication`, `Artifact` and that got me into the weeds of all the Maven stuff which isn't 100% Gradle specific but comes up a lot so I figured out it's ok to include it.

Comment (martinbonnin) [reply README.asciidoc]: Also, I might have messed up with formatting/wrapping. What do you use by default?

Comment (autonomousapps) [reply README.asciidoc]: I manually add line breaks, that's it (:

**Pull request merged**